apps/passport-client/pages/index.tsx

```diff
@@ -115,24 +115,81 @@ class App extends React.Component<object, AppState> {
     } as Partial<AppState>;
   }
 
+  BG_POLL_INTERVAL_MS = 1000 * 60;
+  lastBackgroundPoll = 0;
+  activePollTimout: NodeJS.Timeout | undefined = undefined;
+
   startBackgroundJobs = () => {
-    console.log("Starting background jobs...");
-    this.jobPollUser();
+    console.log("[JOB] Starting background jobs...");
+    document.addEventListener("visibilitychange", () => {
+      this.setupPolling();
+    });
+    this.setupPolling();
   };
 
-  jobPollUser = async () => {
-    console.log("[JOB] polling user");
+  /**
+   * Idempotently enables or disables periodic polling of jobPollServerUpdates,
+   * based on whether the window is visible or invisible.
+   *
+   * If there is an existing poll scheduled, it will not be rescheduled,
+   * but may be cancelled.  If there is no poll scheduled, a new one may be
+   * scheduled.  It may happen immediately after the window becomes visible,
+   * but never less than than BG_POLL_INTERVAL_MS after the previous poll.
+   */
+  setupPolling = async () => {
+    if (!document.hidden) {
+      if (!this.activePollTimout) {
+        const nextPollDelay = Math.max(
+          0,
+          this.lastBackgroundPoll + this.BG_POLL_INTERVAL_MS - Date.now()
+        );
+        this.activePollTimout = setTimeout(
+          this.jobPollServerUpdates,
+          nextPollDelay
+        );
+        console.log(
+          `[JOB] next poll for updates scheduled in ${nextPollDelay}ms`
+        );
+      }
+    } else {
+      if (this.activePollTimout) {
+        clearTimeout(this.activePollTimout);
+        this.activePollTimout = undefined;
+        console.log("[JOB] poll for updates disabled");
+      }
+    }
+  };
+
+  /**
+   * Periodic job for polling the server.  Is scheduled by setupPolling, and
+   * will reschedule itself in the same way.
+   */
+  jobPollServerUpdates = async () => {
+    console.log("[JOB] polling server for updates");
+    this.activePollTimout = undefined;
 
+    // Check for updates to User object.
     try {
       if (this.state?.self) {
         await pollUser(this.state.self, this.dispatch);
       }
     } catch (e) {
-      console.log("[JOB] failed poll user");
-      console.log(e);
+      console.log("[JOB] failed poll user", e);
+    }
+
+    // Trigger sync of E2EE storage, but only if the first-time sync had time
+    // to complete.
+    if (this.state.completedFirstSync) {
+      // TODO(artwyman): Figure out why I need this cast here, and clean up.
+      this.update({ shouldPollDownload: true } as Pick<
+        AppState,
+        keyof AppState
+      >);
```

Comment:
I've run into this before too, I think you can get around it by doing the following: 
```suggestion
      this.update({ ...this.state, shouldPollDownload: true });
```

Reply:
That compiles, but does it change behavior by making the AppState think that all fields are changing rather than only some fields?  I'm not familiar with how the magic of triggering updates when fields change works.